src/condor_collector.V6/offline_plugin.cpp

```diff
@@ -449,7 +449,8 @@ OfflineCollectorPlugin::expire (
 	/* for now, if the ad is of any type other than a startd ad, bail out. currently
 	   absent ads only supported for ads of type Machine, because our offline storage
 	   assumes that. */
-	if ( strcmp(GetMyTypeName(ad),STARTD_ADTYPE) ) {
+	// PRAGMA_REMIND("TJ: how to fix for Slot vs Machine vs StartDaemon ads?");
+	if ( strcmp(GetMyTypeName(ad),STARTD_OLD_ADTYPE) ) {
```

Comment:
Seems like this should include the old ad type (for compatibility) and the new ad type (so that you don't lose any features by upgrading).